src/main/java/teammates/sqllogic/api/SqlEmailGenerator.java

```diff
@@ -0,0 +1,1071 @@
+        String emailBody = Templates.populateTemplate(resendLinksTemplate,
+                "${userName}", SanitizationHelper.sanitizeForHtml(userName),
+                "${userEmail}", userEmail,
+                "${courseName}", SanitizationHelper.sanitizeForHtml(course.getName()),
+                "${courseId}", course.getId(),
+                "${joinFragment}", joinFragmentValue,
+                "${linksFragment}", linksFragmentValue.toString(),
+                "${additionalContactInformation}", additionalContactInformation);
+
+        EmailWrapper email = getEmptyEmailAddressedToEmail(userEmail);
+        email.setContent(emailBody);
+        email.setType(emailType);
+        email.setSubjectFromType(course.getName(), course.getId());
+        return email;
+    }
+
+    /**
+     * Generates for the student an recovery email listing the links to submit/view responses for all feedback sessions
+     * under {@code recoveryEmailAddress} in the past 180 days. If no student with {@code recoveryEmailAddress} is
+     * found, generate an email stating that there is no such student in the system. If no feedback sessions are found,
+     * generate an email stating no feedback sessions found.
+     */
+    public EmailWrapper generateSessionLinksRecoveryEmailForStudent(String recoveryEmailAddress) {
+        List<Student> studentsForEmail = usersLogic.getAllStudentsForEmail(recoveryEmailAddress);
+
+        if (studentsForEmail.isEmpty()) {
+            return generateSessionLinksRecoveryEmailForNonExistentStudent(recoveryEmailAddress);
+        } else {
+            return generateSessionLinksRecoveryEmailForExistingStudent(recoveryEmailAddress, studentsForEmail);
+        }
+    }
+
+    private EmailWrapper generateSessionLinksRecoveryEmailForNonExistentStudent(String recoveryEmailAddress) {
+        String recoveryUrl = Config.getFrontEndAppUrl(Const.WebPageURIs.SESSIONS_LINK_RECOVERY_PAGE).toAbsoluteString();
+        String emailBody = Templates.populateTemplate(
+                EmailTemplates.SESSION_LINKS_RECOVERY_EMAIL_NOT_FOUND,
+                "${userEmail}", SanitizationHelper.sanitizeForHtml(recoveryEmailAddress),
+                "${supportEmail}", Config.SUPPORT_EMAIL,
+                "${teammateHomePageLink}", Config.getFrontEndAppUrl("/").toAbsoluteString(),
+                "${sessionsRecoveryLink}", recoveryUrl);
+        EmailWrapper email = getEmptyEmailAddressedToEmail(recoveryEmailAddress);
+        email.setType(EmailType.SESSION_LINKS_RECOVERY);
+        email.setSubjectFromType();
+        email.setContent(emailBody);
+        return email;
+    }
+
+    private EmailWrapper generateSessionLinksRecoveryEmailForExistingStudent(String recoveryEmailAddress,
+            List<Student> studentsForEmail) {
+        String emailBody;
+
+        var searchStartTime = TimeHelper.getInstantDaysOffsetBeforeNow(SESSION_LINK_RECOVERY_DURATION_IN_DAYS);
```

Comment:
Since you are cleaning this up anyways, let's replace var with the actual type. No real reason to use it in this method.